core/lib/instrument.c

```diff
@@ -6799,7 +6799,7 @@ dr_get_mcontext_priv(dcontext_t *dcontext, dr_mcontext_t *dmc, priv_mcontext_t *
     else if (TEST(DR_MC_CONTROL, dmc->flags))
         dmc->xsp = get_mcontext(dcontext)->xsp;
 
-#ifdef ARM
+#ifdef AARCHXX
     if (TEST(DR_MC_INTEGER, dmc->flags)) {
```

Comment:
I happened to need this locally and it turns out there is a bug here: `dmc` can be NULL, so this should be:
```
    if (mc != NULL || TEST(DR_MC_INTEGER, dmc->flags)) {
```
I hit a crash with just this patch as-is due to this.

Reply:
Ah thanks,
I presume you mean:
`if (dmc != NULL || TEST(DR_MC_INTEGER, dmc->flags)) {`
?

Reply:
No, if `mc` is non-NULL, then there are no flags and everything is filled in and `dmc` is ignored and should be NULL.  Only if `mc` is NULL is `dmc` guaranteed to be non-NULL.

Reply:
Ah okay that makes sense thanks!